updatehub/src/states/prepare_local_install.rs

```diff
@@ -27,6 +27,9 @@ impl StateChangeImpl for State<PrepareLocalInstall> {
     ) -> Result<(StateMachine, actor::StepTransition)> {
         info!("Prepare local install: {:?}", self.0.update_file);
         let dest_path = shared_state.settings.update.download_dir.clone();
+        if !dest_path.exists() {
+            std::fs::create_dir_all(&dest_path)?;
+        }
```

Comment:
The `create_dir_all` already handles the case of the directories exists. So you can avoid the `if` clause here.